File: Data Structures & Algorithms/climbing-stairs/submission-0.cpp
class Solution {
public:
    int climbStairs(int n) {
        vector<int>temp(n+1,0);
        temp[0]=1;
        temp[1]=1;
        for(int i=2;i<=n;i++){
            temp[i]=temp[i-1]+temp[i-2];
        }
        return temp[n];
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res=[]
        nums.sort()
        n=len(nums)
        for i in range(1<<n):
            temp=[]
            valid=True
            for j in range(n):
                if i&(1<<j):
                    if j>0 and nums[j]==nums[j-1] and not (i&(1<<(j-1))):
                        valid=False
                        break
                    temp.append(nums[j])
            if valid:
                res.append(temp)

        return res
                
